### PR TITLE
Exactness criterion

### PR DESCRIPTION
**Pull request by LegendreM**
- [ ] (future PR) Rule 1: Fetch documents that have an attributes that contains exactly the Query and only the Query.
- [x]  Rule 2: Fetch documents that have an attributes starting by the Query.
- [x]  Rule 3: Fetch documents that contains the words of the Query anywhere. 

Related to #165

### CODE REVIEW

File: milli/src/search/criteria/exactness.rs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    Ok(attribute_candidates_array)
+}
+
+fn intersection_of(mut to_intersect: Vec<&RoaringBitmap>) -> RoaringBitmap {
+    match to_intersect.len() {
+        0 => RoaringBitmap::new(),
+        1 => to_intersect[0].clone(),
+        2 => to_intersect[0] & to_intersect[1],
+        _ => {
+            to_intersect.sort_unstable_by(|a, b| a.len().cmp(&b.len()).reverse());
```
Comment (Kerollmops) [inline milli/src/search/criteria/exactness.rs:305]:
```suggestion
            to_intersect.sort_unstable_by_key(|rb| Reverse(rb.len()));
```

You can import `std::cmp::Reverse`.

File: milli/src/search/criteria/mod.rs [hunk 2]
```diff
@@ -182,18 +208,22 @@ impl<'t> CriteriaBuilder<'t> {
...
         for name in self.index.criteria(&self.rtxn)? {
             criterion = match name {
                 Name::Typo => Box::new(Typo::new(self, criterion)),
                 Name::Words => Box::new(Words::new(self, criterion)),
                 Name::Proximity => Box::new(Proximity::new(self, criterion)),
                 Name::Attribute => Box::new(Attribute::new(self, criterion)),
+                Name::Exactness => Box::new(Exactness::new(self, criterion, &primitive_query)?),
                 Name::Asc(field) => Box::new(AscDesc::asc(&self.index, &self.rtxn, criterion, field)?),
                 Name::Desc(field) => Box::new(AscDesc::desc(&self.index, &self.rtxn, criterion, field)?),
                 _otherwise => criterion,
```
Comment (Kerollmops) [inline milli/src/search/criteria/mod.rs:229]:
```suggestion
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Please, fix the final warnings. I don't quite like the `intersection_of` implementation, but... I don't have the time to implement it :)

File: milli/src/search/criteria/exactness.rs [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                        if candidates.is_empty() { break; }
+                        candidates &= bitmap;
+                    }
+
+                    candidates
+                },
+            }
+        }
+    }
+}
```
Comment (Kerollmops) [inline milli/src/search/criteria/exactness.rs:321]:
```suggestion
fn intersection_of(rbs: &mut [RoaringBitmap]) -> RoaringBitmap {
    rbs.sort_unstable_by_key(RoaringBitmap::len);

    let mut iter = rbs.iter();
    match iter.next() {
        Some(first) => iter.fold(first.clone(), |acc, rb| acc & rb),
        None => RoaringBitmap::new(),
    }
}
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: milli/src/search/criteria/exactness.rs (hunk 3 above)
Comment (LegendreM) [reply milli/src/search/criteria/exactness.rs]: You're right, this is completelly better!

**Comment by Kerollmops** [comment]
Next time, @LegendreM, please merge by using bors :)

**Pull request merged**